src/Microsoft.Kusto.ServiceLayer/QueryExecution/ResultSet.cs

```diff
@@ -658,12 +625,30 @@ private async Task SendCurrentResults()
                 }
             }
             finally
-            { 
+            {
+                // and finally:
+                // Make a call to send ResultCompletion and await its completion. This is just for backward compatibility with older protocol
+                //
+                await (ResultCompletion?.Invoke(this) ?? Task.CompletedTask);
+
                 // Release the sendResultsSemphore so the next invocation gets unblocked
                 //
                 sendResultsSemphore.Release();
             }
         }
+        #endregion
+
+        #region Private Helper Methods
+        /// <summary>
+        /// Sends the ResultsUpdated message if the number of rows has changed since last send.
+        /// </summary>
+        /// <param name="stateInfo"></param>
+        private void SendResultAvailableOrUpdated(object stateInfo = null)
```

Comment:
Is this `stateInfo` still needed?

Reply:
Yes it is a TimerCallback

namespace System.Threading
{
    //
    // Summary:
    //     Represents the method that handles calls from a System.Threading.Timer.
    //
    // Parameters:
    //   state:
    //     An object containing application-specific information relevant to the method
    //     invoked by this delegate, or null.
    public delegate void TimerCallback(object? state);
}